state-chain/pallets/cf-threshold-signature/src/lib.rs

```diff
@@ -810,14 +806,11 @@ where
 	type ValidatorId = T::ValidatorId;
 
 	fn request_signature(payload: PayloadFor<T, I>) -> RequestId {
-		let request_type = if <T::TargetChainCrypto as ChainCrypto>::sign_with_specific_key() {
-			T::KeyProvider::active_epoch_key().defensive_map_or_else(
-				|| RequestType::CurrentKey,
-				|EpochKey { key, epoch_index, .. }| RequestType::SpecificKey(key, epoch_index),
-			)
-		} else {
-			RequestType::CurrentKey
-		};
+		let request_type = T::KeyProvider::active_epoch_key().defensive_map_or_else(
+			|| RequestType::SpecificKey(Default::default(), Default::default()),
```

Comment:
If we use `RequestType::CurrentKey` it will fail more gracefully (it will emit `CurrentKeyUnavailable` and retry with CurrentKey until one is available).

Reply:
I have removed the ```CurrentKey``` variant in the ```RequestType``` since we dont use it anymore. The ceremonies will always be signed with key active at the time of threshold signature request since do optimistic rotation for all chains now and every ceremony has a key associated to it.